src/foam/swift/refines/Method.js

```diff
@@ -293,3 +293,31 @@ return MInfo(classInfo())
     }
   ],
 });
+
+foam.CLASS({
+  refines: 'foam.core.AbstractMethod',
+  package: 'foam.swift.refines',
+  name: 'CreateChildRefines',
+  flags: ['swift'],
+  methods: [
+    function createChildMethod_(child) {
+      var result = child.clone();
+      var props = child.cls_.getAxiomsByClass(foam.core.Property);
+      for ( var i = 0 ; i < props.length ; i++ ) {
+        var prop = props[i];
+        if ( this.hasOwnProperty(prop.name) && ! child.hasOwnProperty(prop.name) ) {
```

Comment:
I think this is the more correct behaviour because, without the this.hasOwnProperty check, we'll be setting javaCode on children whose parents don't have it set. Removing this check breaks java because there's a few models out there that don't implement all methods on the interface and if javaCode isn't set, the method doesn't get generated and the compiler complains.

Reply:
The reason java breaks is because there were some models without javaCode on all methods of the interface they implement. An alternative to this PR is here: https://github.com/foam-framework/foam2/pull/1407